samples/file-encryptor/enclave/openssl_src/keys.cpp

```diff
@@ -0,0 +1,435 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openssl/rand.h>
+#include <string.h>
+
+#include "common/trace.h"
+#include "encryptor.h"
+
+#define ENCRYPT_OPERATION true
+#define DECRYPT_OPERATION false
+
+void ecall_dispatcher::dump_data(
+    const char* name,
+    unsigned char* data,
+    size_t data_size)
+{
+    TRACE_ENCLAVE("Data name: %s", name);
+    for (size_t i = 0; i < data_size; i++)
+    {
+        TRACE_ENCLAVE("[%ld]-0x%02X", i, data[i]);
+    }
+    TRACE_ENCLAVE("\n");
+}
+
+// Compute the sha256 hash of given data.
+int ecall_dispatcher::Sha256(
+    const uint8_t* data,
+    size_t data_size,
+    uint8_t sha256[32])
+{
+    int ret = -1;
```

Comment:
It seems that the mbedtls implementation uses 1 to indicate the failure while now here uses -1. Can you double check for the consistency?

Reply:
updated to use 1 for error, 0 for success.